master/internal/configpolicy/task_config_policy_intg_test.go

```diff
@@ -46,9 +49,144 @@ func TestPriorityAllowed(t *testing.T) {
 	require.False(t, ok)
 }
 
-func addWorkspacePriorityLimit(t *testing.T, pgDB *db.PgDB, limit int) model.Workspace {
-	ctx := context.Background()
+func TestValidateNTSCConstraints(t *testing.T) {
+	require.NoError(t, etc.SetRootPath(db.RootFromDB))
+	pgDB, cleanup := db.MustResolveNewPostgresDatabase(t)
+	defer cleanup()
+	db.MustMigrateTestPostgres(t, pgDB, db.MigrationsFromDB)
+
+	wkspPriorityLimit := 7
+	user := db.RequireMockUser(t, pgDB)
+
+	t.Run("no constraints set - ok", func(t *testing.T) {
+		resourceManager := mocks.ResourceManager{}
+		resourceManager.On("SmallerValueIsHigherPriority", mock.Anything).Return(false, nil)
+
+		config := defaultConfig()
+		ok, err := ValidateNTSCConstraints(context.Background(), 1, config, &resourceManager)
+		require.NoError(t, err)
+		require.True(t, ok)
+	})
+
+	t.Run("running in wksp with constraints - not ok", func(t *testing.T) {
+		w := addWorkspacePriorityLimit(t, user, wkspPriorityLimit)
+		resourceManager := mocks.ResourceManager{}
+		resourceManager.On("SmallerValueIsHigherPriority", mock.Anything).Return(false, nil)
+
+		config := defaultConfig()
+		ok, err := ValidateNTSCConstraints(context.Background(), w.ID, config, &resourceManager)
+		require.False(t, ok)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "requested priority")
+	})
+
+	t.Run("running in wksp without constraints - ok", func(t *testing.T) {
+		resourceManager := mocks.ResourceManager{}
+		resourceManager.On("SmallerValueIsHigherPriority", mock.Anything).Return(false, nil)
+		w := model.Workspace{Name: uuid.NewString(), UserID: user.ID}
+		_, err := db.Bun().NewInsert().Model(&w).Exec(context.Background())
```

Comment:
Random thought I'm interested in your opinion on: I wonder if we should use `context.TODO()` here. Maybe someday we want to do something like

```go
func TestWhatever(t *testing.T) {
	ctx, cancel := context.WithCancel(context.Background())
	defer cancel()

	result, err := FunctionUnderTest(ctx)
```
I'm completely okay with this being `context.Background()`; I'm mostly just pondering.

Reply:
I like the idea of using `context.WithCancel`! I remember a PR comment once that stated a preference for `context.Background()` instead of `context.TODO()` in tests to make it easier to find the "genuine" to dos. I don't remember who left the comment, and I'm sure I've forgotten some nuance, but I liked the reasoning.